cranelift/control/src/chaos.rs

```diff
@@ -23,6 +28,35 @@ impl ControlPlane {
     /// pseudo-random data is exhausted or the control plane was constructed
     /// with `default`.
     pub fn get_decision(&mut self) -> bool {
-        self.data.pop().unwrap_or_default()
+        self.data.pop().unwrap_or_default() & 1 == 1
+    }
+
+    /// Shuffles the items in the slice into a pseudo-random permutation if
+    /// the control plane was constructed with `arbitrary`.
+    ///
+    /// The default operation, to leave the slice unchanged, will always be
+    /// performed if the pseudo-random data is exhausted or the control
+    /// plane was constructed with `default`.
```

Comment:
Can we add an early-out conditional to `shuffle` along the lines of what this comment describes (if data is exhausted ...)? The effect is the same now but we construct an `Unstructured`, take a slice, and do some other work before we hit the `break` with the current code.